### PR TITLE
:createstat, :deletestat, and :deletestats

### PR DESCRIPTION
**Pull request by WalkerOfBacon**
:createstat creates a new leaderstat
:deletestat removes an existing leaderstat
:deletestats removes all existing leaderstats

### CODE REVIEW

**Comment by NoobBucket** [comment]
There is already a command to remove all leaderstats

**Comment by WalkerOfBacon** [comment]
> There is already a command to remove all leaderstats

Which one?

**Comment by WalkerOfBacon** [comment]
I checked all the command and there isn't one that does the same thing

**Comment by NoobBucket** [comment]
`:resetstats` sets the target players leaderstats to 0

**Comment by WalkerOfBacon** [comment]
> `:resetstats` sets the target players leaderstats to 0

Im using `:removestats` not `:resetstats`

File: MainModule/Server/Commands/Moderators.lua [hunk 1]
```diff
@@ -4927,6 +4927,87 @@ return function(Vargs, env)
...
+			AdminLevel = "Moderators";
+			Function = function(plr: Player, args: {string})
+
+				local statName = assert(args[1], "Missing stat name")
+
+				for _, v in service.GetPlayers(plr, args[1]) do
+					local leaderstats = plr:FindFirstChild("leaderstats")
+
+					if leaderstats and (leaderstats:IsA("Folder") or leaderstats:IsA("StringValue")) then
+						local statName = Instance.new("IntValue")
```
Comment (Expertcoderz) [inline MainModule/Server/Commands/Moderators.lua:4945]: Prefer `service.New` over `Instance.new` in Adonis code:
```lua
service.New("IntValue", {
    Parent = leaderstats,
    Name = args[1],
    Value = 0
}
```

Comment (WalkerOfBacon) [reply MainModule/Server/Commands/Moderators.lua:4945]: Will do

**Comment by WalkerOfBacon** [comment]
All good now

File: MainModule/Server/Commands/Moderators.lua [hunk 2]
```diff
@@ -4928,6 +4928,90 @@ return function(Vargs, env)
...
+			end
+		};
+
+		DeleteAllStats = {
+			Prefix = Settings.Prefix;
+			Commands = {"deletestats", "dstats"};
+			Description = "Deletes all the leaderstats";
+			AdminLevel = "Moderators";
+			Function = function(plr)
+				local thestats = plr.leaderstats:GetChildren()
```
[the remark this answers is not in the document]
Comment (WalkerOfBacon) [reply MainModule/Server/Commands/Moderators.lua:5003]: I will get to fixing that at some point

File: MainModule/Server/Commands/Moderators.lua [hunk 3]
```diff
@@ -4928,6 +4928,90 @@ return function(Vargs, env)
...
+			Commands = {"deletestat", "dstat"};
+			Args = {"name"};
+			Description = "Deletes a leaderstat";
+			AdminLevel = "Moderators";
+			Function = function(plr: Player, args: {string})
+				local statName = assert(args[1], "Missing the stat name!")
+
+				for _, v in service.GetPlayers(plr, args[1]) do
+					local children = plr:GetChildren()
+					local leaderstats = plr.leaderstats -- This checks if there is a leaderstats folder/string value [Don't change it unless you know what you are doing]
```
Comment (Sceleratis) [inline MainModule/Server/Commands/Moderators.lua:4977]: Another issue. This won't actually check if leaderstats exists... it's just going to error and break the entire loop/command if it *doesn't* exist already. That's aside from the issue that it's currently running this on the player who ran the command with a number of loops equal to the number of players that it should have been ran on...

Comment (WalkerOfBacon) [reply MainModule/Server/Commands/Moderators.lua]: I just fixed this

**Comment by WalkerOfBacon** [comment]
fix

**Pull request closed without merging**